crates/cargo-bsp/src/cargo_communication/cargo_messages_handle.rs

```diff
@@ -198,7 +226,21 @@ where
     }
 
     fn report_suite_finished(&self, task_id: TaskId, result: SuiteResults) {
-        self.report_task_finish(task_id, StatusCode::Ok, None, Some(result.to_test_report()))
+        if let TaskState::Test(test_state) = &self.state.task_state {
+            let tested_target = match test_state.current_build_target.clone() {
+                Some(t) => t,
+                None => {
+                    warn!("No target is currently tested");
+                    return;
+                }
+            };
+            self.report_task_finish(
+                task_id,
+                StatusCode::Ok,
```

Comment:
Zawsze tutaj dajemy ok nawet jak  przy SuiteEvent::Failed?

Reply:
Status code w taskach oznacza, czy wykonanie taska się powiodło. W tym przypadku task to "wykonaj testy", co się powiodło, bo wszystkie testy zostały wykonane. A to, że jakiś test zakończył się z błędem nic do tego nie ma (choć jest raportowane potem w data w task finish).
Dość dobrym przykładem na ogarnięcie tego jest task kompilacji i task z unit graphem. Task kompilacji to task "wykonaj kompilację" (więc też zawsze kończymy ok, nawet jak exit code będzie niezerowy). Natomiast task z unit graphem to "zdobądź unit graph", a nie "wykonaj komendę z unit graphem", dlatego np jak nie zdobędziemy unit graphu, to dajemy status code error.